src/shogun/preprocessor/FisherLDA.cpp

```diff
@@ -250,7 +251,10 @@ bool CFisherLDA::fit(CFeatures *features, CLabels *labels, int32_t num_dimension
 		// x=M
 		// MatrixXd M=Sw.householderQr().solve(Sb);
 		// calculate the eigenvalues and eigenvectors of M.
-		EigenSolver<MatrixXd> es(Sw.householderQr().solve(Sb));
+		SGMatrix<float64_t> M=
+				linalg::qr_solver(SGMatrix<float64_t>(Sw), SGMatrix<float64_t>(Sb));
+		SGMatrix<float64_t>::EigenMatrixXtMap M_eig=M;
+		EigenSolver<MatrixXd> es(M_eig);
 
```

Comment:
Cool, the implicit cast to `SGMatrix` works but, at least in this case, the other way around does not: `EigenSolver<T>` doesn't accept a `Map` type and I can't find a way to avoid the explicit cast.